znai-cli/src/main/java/com/twosigma/znai/cli/ZnaiCliConfig.java

```diff
@@ -214,7 +219,8 @@ private void parseArgs(String[] args) {
 
         markupType = commandLine.hasOption(MARKUP_TYPE_KEY) ? commandLine.getOptionValue(MARKUP_TYPE_KEY) : MarkupTypes.MARKDOWN;
 
-        sourceRoot = Paths.get(commandLine.hasOption(SOURCE_KEY) ? commandLine.getOptionValue(SOURCE_KEY) : "")
+        isSourceRootSet = commandLine.hasOption(SOURCE_KEY);
+        sourceRoot = Paths.get(isSourceRootSet ? commandLine.getOptionValue(SOURCE_KEY) : "")
```

Comment:
Actually you can decide `sourceRoot` here, so you don't have to add `isSourceRootSet` as a member variable.